src/utils/duration.ts

```diff
@@ -0,0 +1,23 @@
+import assert from "assert";
+
+/**
+ * Formats duration with a format like "2h 25m 12s".
+ * @param start timestamp in milliseconds
+ * @param end timestamp in milliseconds
+ */
+export const formatDuration = (start: number, end: number): string => {
+  const totalSeconds = (end - start) / 1000;
+  assert(totalSeconds >= 0, "Cannot format negative duration");
+
+  const hours = Math.floor(totalSeconds / 3600);
```

Comment:
Shouldn't we start with day? (Our hard limit is 7d)

Reply:
Agreed.

Also noticed that we can have weird results when one of the values is 0; e.g., for exactly 23 hours, we currently print `23h 0s` rather than `23h 0m 0s` (or even just `23h`). I think I'll change that so it would be `23h 0m 0s`, basically only ignore the _leading_ zeros.